phaser/engines/gradient/run.py

```diff
@@ -285,6 +289,11 @@ def run_engine(args: EngineArgs, props: GradientEnginePlan) -> ReconsState:
                 state, iter_solver_states[sol_i], filter_vars(iter_grads, solver.params), losses['total_loss']
             )
             state = apply_update(state, update)
+            update_mag = xp.linalg.norm(update['positions'], axis=-1)
```

Comment:
What happens if no position solver is specified, and 'positions' is not in update?